var/spack/repos/builtin/packages/snappy/package.py

```diff
@@ -33,14 +33,26 @@ class Snappy(CMakePackage):
 
     version('1.1.7', 'ee9086291c9ae8deb4dac5e0b85bf54a')
 
+    variant('shared', default=True, description='Build shared libraries')
+
+    def cmake_args(self):
+        spec = self.spec
+
+        args = [
+            '-DBUILD_SHARED_LIBS:BOOL={0}'.format(
+                'ON' if '+shared' in spec else 'OFF')
+        ]
+
+        return args
+
     @run_after('install')
     def install_pkgconfig(self):
         mkdirp(self.prefix.lib.pkgconfig)
 
         with open(join_path(self.prefix.lib.pkgconfig, 'snappy.pc'), 'w') as f:
             f.write('prefix={0}\n'.format(self.prefix))
             f.write('exec_prefix=${prefix}\n')
-            f.write('libdir={0}\n'.format(self.prefix.lib))
```

Comment:
We usually install into ‘lib’, if dependents need to know the location of the libraries, they need to be configured appropriately